tests/factories.py

```diff
@@ -114,13 +114,13 @@ class ServerScanResultFactory:
     @staticmethod
     def create(
         server_info: ServerConnectivityInfo = ServerConnectivityInfoFactory.create(),
-        scan_commands_results: Optional[ScanCommandResultsDict] = None,
+        scan_commands_results: Optional[ScanCommandResults] = None,
         scan_commands_errors: Optional[ScanCommandErrorsDict] = None,
     ) -> ServerScanResult:
-        final_results: ScanCommandResultsDict = (
+        final_results: ScanCommandResults = (
             scan_commands_results
             if scan_commands_results
-            else {ScanCommand.TLS_COMPRESSION: CompressionScanResult(supports_compression=True)}
+            else ScanCommandResults(tls_compression=CompressionScanResult(supports_compression=True))
```

Comment:
I did not see any way of using the constant as key. Any suggestion would be welcome.